libexec/bats-core/bats

```diff
@@ -200,12 +200,19 @@ while [[ "$#" -ne 0 ]]; do
   shift
 done
 
-if [[ -d "$BATS_RUN_TMPDIR" ]]; then
-  printf "Error: BATS_RUN_TMPDIR (%s) already exists\n" "$BATS_RUN_TMPDIR" >&2
-  printf "Reusing old run directories can lead to unexpected results ... aborting!\n" >&2
-  exit 1
+if [[ -n "${BATS_RUN_TMPDIR:-}" ]];then
```

Comment:
So far, we don't allow for setting `BATS_RUN_TMPDIR` from outside. It is already customizable via `BATS_TMPDIR`, so I think we don't need this additional complexity here.

Reply:
Do you think having this check (for `-n "${BATS_RUN_TMPDIR:-}") here anyway could potentially expose and add a guardrail against race conditions?

Reply:
It would allow an opportunity for better reporting in cases where it is detectable but I don't think it can eliminate race conditions entirely, since the check and `mktemp` are not one atomic operation.